### PR TITLE
Upgrade to Svelte v3

### PR DESCRIPTION
**Pull request by antony**
This is a work in progress, but I might need some help.

The goal here is to shamelessly clone the changes from https://github.com/rollup/rollup-plugin-svelte/commit/ba9bd2c99d92c925c5034f68e9d33ee075e356da - there are only two - to make this plugin work with both Svelte 2 and 3.

Here's a test component, which I'm using to test the change. https://v3.svelte.technology/repl?version=3.0.0-beta.3&gist=cd86b9640ad3f79eceef41bd4f726906

The issue I'm currently encountering is this one:

screenshot from 2019-02-15 08-28-48

Also - sorry about the changelog.md? I've no idea what's going on with that or what I'm supposed to do with it :)

### CODE REVIEW

File: src/svelte-asset.js [hunk 1]
```diff
@@ -39,7 +41,8 @@ class SvelteAsset extends Asset {
       generate: 'dom',
       format: 'cjs',
```
Comment (DeMoorJasper) [inline src/svelte-asset.js]: Perhaps we should also use `major_version >= 3 ? 'esm' : 'es';` ?

File: src/svelte-asset.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
 const path = require('path');
-const {compile, preprocess} = require('svelte');
 const {Asset} = require('./parcel-adapter');
 const {sanitize, capitalize} = require('./utils');
 
+const {version: svelteVersion} = require('svelte/package.json');
```
Comment (DeMoorJasper) [inline src/svelte-asset.js]: Perhaps we should extract a major version from this similar to how it works in the commit you linked

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,144 +4,22 @@ All notable changes to this project will be documented in this file. Dates are d
 
 Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).
 
-#### [1.2.2](https://github.com/DeMoorJasper/parcel-plugin-svelte/compare/1.2.1...1.2.2)
```
Comment (DeMoorJasper) [inline CHANGELOG.md]: I'm gonna get rid of the changelog, it's not that usefull imo

File: src/svelte-asset.js (hunk 2 above)
Comment (antony) [reply src/svelte-asset.js]: Good point - I had intended to grab the major version :)

File: src/svelte-asset.js (hunk 1 above)
Comment (antony) [reply src/svelte-asset.js]: If it works yeah - that is correct.

File: CHANGELOG.md (hunk 3 above)
Comment (antony) [reply CHANGELOG.md]: Fair enough!

**Comment by DeMoorJasper** [comment]
Thanks for the amazing example on how to implement Svelte 3 support

**Pull request closed without merging**